### ISSUE TITLE
Subscripts and superscripts are incorrectly sized inside SVG when `font-size` is set on `<foreignObject>`

### ISSUE THREAD

**Issue by rnwst** [mathjax/MathJax#2975]
### Issue Summary

When `style="font-size: ..."` is set on a `<foreignObject>` tag inside of an SVG, subscripts and superscripts are incorrectly sized (see the screenshot below).
![image](https://user-images.githubusercontent.com/98110034/205403103-8f3e346f-df98-4bf0-9292-86d858a97595.png)
When no `style="font-size: ..."` is set on the `<foreignObject>`, this bug does not appear. The problem exists both for SVG as well as CHTML output, although the size and positioning appears to be slightly different in each case.

### Steps to Reproduce:

1. Open the HTML below (or open this [fiddle](https://jsfiddle.net/np8fo4j2/)).
2. Observe subscript and superscript sizing inside of SVG.

### Technical details:

* MathJax Version: 3.2.2
* Client OS: Linux 64-bit
* Browsers:
  - Chromium 107.0.5304.121 
  - Firefox 107.0
  - Microsoft Edge 106.0.1370.47

I am using the following MathJax configuration: None

and loading MathJax via

``` html
<script async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"></script>
```

### Supporting information:

HTML file:
```html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"></script>
</head>
<body>
<div style="font-size: 40px">Outside SVG: \(\int_a^b f(x)dx\)</div>
<svg xmlns="http://www.w3.org/2000/svg" width="650px" height="550px" viewBox="0 0 65 55">
<foreignObject x="0" y="5" width="65" height="55" style="font-size:4px"><div>Inside SVG: \(\int_a^b f(x)dx\)</div>
</foreignObject>
</svg>
</body>
</html>
```

**Comment by dpvc** [comment]
This is because the default value for the `scriptminsize` attribute in MathML is set at `8px`, but since the font size is 4px, that means the super- and subscripts will end up being twice as big as the standard text.  The MathML specification says it should be `8pt`, but I ended up switching to `8px` somewhere along the line because 8pt was too big: there are 10pt to an em, and so 8pt is too large in most situations, and it seemed better to have the minimum based on screen pixels in order to avoid too small fonts.  But since `px` are no longer actual pixels, but relative to other scaling factors, as in your example, then that no longer does what it should.

So you might want to set the default minimum script size to something smaller.  For example,

``` js
MathJax = {
  startup: {
    ready() {
      MathJax._.core.MmlTree.MmlNodes.math.MmlMath.defaults.scriptminsize = '8pt';
      MathJax.startup.defaultReady();
    }
  }
}
```

sets it back to `8pt`, which works for your example, but may not work for a real page.  You could use `1px` or something like that, if that works better.

**Comment by rnwst** [comment]
Many thanks @dpvc for your quick response, the detailed explanation, and the fix!

While I now understand the reasons for this, I still consider this a bug, since the 'actual font-size' of the sub- and superscripts is much larger than `8px`. This is, as you have pointed out, because 'SVG `px`' are not equal to 'normal `px`'. A simple fix for this appears to be to set `scriptminsize` in `pt` instead of `px` (for some reason, subscripts and superscripts inside and outside of the SVG then appear to be the same size, even when their size is limited by `scriptminsize` - do you know why?). 

Would you consider changing the default value of `scriptminsize` from `8px` to `6pt` or are there other side-effects I haven't considered?

**Comment by dpvc** [comment]
Yes, the value used for `scriptminsize` should be reconsidered.  Since it is already changed from the MathML specification, it should be changed to a value that works in more situations.

**Comment by dpvc** [comment]
I've made a PR to improve the handling of `scriptminsize` and to use a better default value.
